src/elm/interval.coffee

```diff
@@ -180,14 +180,21 @@ module.exports.Collapse = class Collapse extends Expression
       # width equal to the result of the successor function for the point type).
       if !perWidth?
         if intervalsClone[0].low?
-          if intervalsClone[0].low.constructor.name == 'DateTime'
-            # Bonnie will always use milliseconds
-            # TODO determine this dynamically using duration between low and successor
-            perWidth = new Quantity(value: 1, unit: 'millisecond')
+          if intervalsClone[0].low.isDateTime
+            precisionUnits = intervalsClone[0].low.getPrecision()
+            perWidth = new Quantity(value: 1, unit: precisionUnits)
           else if intervalsClone[0].low.isQuantity
             perWidth = doSubtraction(successor(intervalsClone[0].low), intervalsClone[0].low)
           else
             perWidth = successor(intervalsClone[0].low) - intervalsClone[0].low
+        else if intervalsClone[0].high?
+          if intervalsClone[0].high.isDateTime
+            precisionUnits = intervalsClone[0].high.getPrecision()
+            perWidth = new Quantity(value: 1, unit: precisionUnits)
+          else if intervalsClone[0].high.isQuantity
+            perWidth = doSubtraction(predecessor(intervalsClone[0].high), intervalsClone[0].high)
```

Comment:
I'm confused.  Isn't this going to give you a negative number for the width?  Is that what you want?

E.g., predecessor of `5 'cm'` is `4 'cm'`, so `predecessor(5 'cm') - 5 'cm'` is `4 'cm' - 5 'cm'` is `-1 'cm'`.

Reply:
good point. I'll fix this